azure-maven-plugin/src/main/java/com/microsoft/azure/plugin/login/SelectSubscriptionMojo.java

```diff
@@ -23,11 +35,102 @@
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        final Log log = getLog();
+        final AzureTokenCredentials tokenCredentials;
+
+        if (!AzureAuthHelper.existsAzureSecretFile()) {
+            throw new MojoFailureException("You are not logged in.");
+        }
+
+        try {
+            tokenCredentials = AzureAuthHelper.getMavenAzureLoginCredentials();
+        } catch (IOException e) {
+            throw new MojoFailureException("Cannot read azure credentials due to error: " + e.getMessage());
+        }
+
+        final Authenticated azure = Azure.configure().authenticate(tokenCredentials);
+        Subscription selectSubscription = null;
+
+        final PagedList<Subscription> subscriptions = azure.subscriptions().list();
+        for (final Subscription subscription : subscriptions) {
+            if (StringUtils.equalsIgnoreCase(subscriptionId, subscription.subscriptionId()) ||
+                    StringUtils.equalsIgnoreCase(subscriptionId, subscription.displayName())) {
+                selectSubscription = subscription;
+                break;
+            }
+        }
+
+        final String oldSelectedSubscription = tokenCredentials.defaultSubscriptionId();
+
+        if (selectSubscription == null) {
+            if (StringUtils.isNotBlank(subscriptionId)) {
+                throw new MojoFailureException(String.format("The subscription of '%s' doesn't exist.", subscriptionId));
+            }
+            if (subscriptions.size() == 0) {
+                throw new MojoExecutionException("No subscriptions are found.");
+            } else if (subscriptions.size() == 1) {
+                selectSubscription = subscriptions.get(0);
+            } else {
+                // promote user for select subscription
+                // TODO: wrap it in an utility method
+                System.out.println("Please choose from the following subscriptions:");
+                int index = 1;
+                for (final Subscription subscription : subscriptions) {
+                    final boolean current = StringUtils.equalsIgnoreCase(oldSelectedSubscription, subscription.subscriptionId());
+                    final String subscriptionLine = String.format("%2d. %s (%s)%s", index++, subscription.displayName(), subscription.subscriptionId(),
+                            current ? " [CURRENT]" : "");
+
+                    if (current) {
+                        System.out.println(TextUtils.green(subscriptionLine));
+                    } else {
+                        System.out.println(subscriptionLine);
+                    }
+
+                }
+
+                final Scanner scanner = getScanner();
+                int userSelected = -1;
+                while (true) {
+                    System.out.printf("Enter index value for subscription id: ");
+                    System.out.flush();
+                    final String input = scanner.nextLine();
```

Comment:
`scanner.nextInt()`?

Reply:
https://stackoverflow.com/questions/13102045/scanner-is-skipping-nextline-after-using-next-or-nextfoo

nextInt will introduce strange codes